core/shared/src/main/scala/scalaz/zio/Fiber.scala

```diff
@@ -51,6 +51,17 @@ trait Fiber[+E, +A] { self =>
    */
   def interrupt0(ts: List[Throwable]): IO[Nothing, Unit]
 
+  /**
+   * Add an exit handler for when the fiber terminates and receive information
+   * on whether the fiber terminated normally, with unhandled error, or with
+   * exception.
+   *
+   * The specified action will be invoked after the fiber has finished running
+   * (including all finalizers). If the specified action throws an exception,
+   * it will be reported to the parent fiber's unhandled error handler.
+   */
+  def finished(f: ExitResult[E, A] => IO[Nothing, Unit]): IO[Nothing, Unit]
```

Comment:
Usually, this is called something like `onComplete`. May be worth considering.

Reply:
I'm OK with `onComplete`. No big preference either way.